feat: watch next and sort different

pauldambra opened the pull request:
# home page watch next

<img width="429" alt="Screenshot 2024-10-21 at 13 47 15" src="https://github.com/user-attachments/assets/8aa14e64-6ab6-4ccc-939a-4722d9a6a3b9">

folk don't click on the recording links on the home page 
let's show them something different and see what they do

# order by

<img width="337" alt="Screenshot 2024-10-21 at 13 51 50" src="https://github.com/user-attachments/assets/2e55976a-2f21-4b19-9993-1eb3a5e01db8">

we've seen. multiple instances of folk being confused by sorting
both in tickets and when we tried to experiment with it

i have multiple times forgotten that I set the sort order and then been confused and I know how it works

so let's put the sorting control right next to the list

pauldambra committed:
first pass

pauldambra committed:
second pass

pauldambra committed:
third pass

pauldambra commented on frontend/src/lib/components/Playlist/Playlist.tsx:
```diff
@@ -44,6 +44,7 @@ export type PlaylistProps<T> = {
     onChangeSections?: (activeKeys: string[]) => void
     'data-attr'?: string
     activeItemId?: string
+    controls?: JSX.Element | null
```
playlist is used in multiple places so keeping this row relatively abstract as "controls" rather than just calling it sorting

pauldambra commented on frontend/src/scenes/project-homepage/WatchNextPanel.tsx:
```diff
@@ -28,7 +32,7 @@ export function RecordingRow({ recording }: RecordingRowProps): JSX.Element {
         <ProjectHomePageCompactListItem
             title={asDisplay(recording.person)}
-            subtitle={`Recorded ${dayjs(recording.start_time).fromNow()}`}
+            subtitle={`activity score: ${recording.mouse_activity_count}`}
```
wanted to keep this vague so went with "activity score"
thought it might sound exciting

pauldambra commented on frontend/src/scenes/session-recordings/filters/OrderingFilters.tsx:
unused component

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

pauldambra committed:
no need for border when no controls

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

pauldambra commented on frontend/src/lib/constants.tsx:
```diff
@@ -215,7 +215,6 @@ export const FEATURE_FLAGS = {
     INSIGHT_VARIABLES: 'insight_variables', // owner: @Gilbert09 #team-data-warehouse
     WEB_EXPERIMENTS: 'web-experiments', // owner: @team-feature-success
     BIGQUERY_DWH: 'bigquery-dwh', // owner: @Gilbert09 #team-data-warehouse
-    REPLAY_DEFAULT_SORT_ORDER_EXPERIMENT: 'replay-order-by-experiment', // owner: #team-replay
```
@richard-better i removed the experiment rather than maintain it through the refactor - i think the feedback on my idea showed it was confusing enough that we should try a different tack

pauldambra committed:
fix

daibhin commented on frontend/src/scenes/session-recordings/playlist/SessionRecordingsPlaylist.tsx:
```diff
@@ -23,6 +23,70 @@ import {
+    filters?: RecordingUniversalFilters
+    setFilters?: (filters: Partial<RecordingUniversalFilters>) => void
+}): JSX.Element | null {
+    return filters && setFilters ? (
```
Personal preference but I typically do this type of checking outside the component and make the props mandatory. It can be confusing when reading code that implements this component to understand why it isn't rendering without looking at its implementation otherwise

pauldambra replied:
yeah, i go backwards and forwards on this.

i feel like a lot of react components are hard to read because we mix presentation and arrangement

i wish sometimes we had things like

```
<replay scene>
 <filters/>
 <playlist/>
 <player/>
 <inspector/>
</replay scene>
```

but totally agree that means you have to read every component

and in this case agree it does feel like passing my trousers to the shopkeeper so they can find my wallet to take payment

daibhin replied:
That is the most brilliant and terrifying Paul analogy yet 😂

daibhin reviewed (approved):
Agreed that this should live within the playlist. Ideally we wouldn't have to introduce a new row to accommodate it but any time I've previously considered this it feels like it would be too hidden behind the cog menu.

Let's go with this for now and review as part of the Replay offsite

pauldambra commented on frontend/src/scenes/project-homepage/WatchNextPanel.tsx:
```diff
@@ -28,7 +32,7 @@ export function RecordingRow({ recording }: RecordingRowProps): JSX.Element {
         <ProjectHomePageCompactListItem
             title={asDisplay(recording.person)}
-            subtitle={`Recorded ${dayjs(recording.start_time).fromNow()}`}
+            subtitle={`activity score: ${recording.mouse_activity_count}`}
```
yeah, wondered this too... and justifies it not being simple mouse activity 👍

pauldambra committed:
fiddling

pauldambra committed:
Merge branch 'master' into feat/watch-next

pauldambra committed:
schema

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

Twixes committed:
Put the icon on the right

Tooltip icons on tooltips should be on the right, they're not so important to be the first thing you see

pauldambra committed:
Merge branch 'master' into feat/watch-next

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

pauldambra committed:
fix

github-actions[bot] committed:
Update query snapshots

pauldambra committed:
Merge branch 'master' into feat/watch-next

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

pauldambra committed:
Merge branch 'master' into feat/watch-next

pauldambra committed:
snapshots

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)
